### DISCUSSION TITLE
Parallel routes and dynamic routes

### DISCUSSION THREAD

**Discussion by sergio-milu** [vercel/next.js#60884] [App Router]
Hello everyone,

I have a question regarding parallel routes and dynamic routes, the use case is this

I have an app that has a sidebar and a main section and the layout and folder structure looks like this

```
/app
    [userId]
        @sidebar
            notifications
                [[...id]]  <- dynamic segment optiona catch all
                    page.tsx
        notifications
            [id]
                page.tsx
            pages.tsx
        page.tsx
```
        
 routes could be

`/123/notifications`
or
`/123/notifications/456`

this seems legit to me as all these routes can be reached, thing is in build time i'm getting this error

`Error: You cannot define a route with the same specificity as a optional catch-all route ("/[userId]/notifications" and "/[userId]/notifications[[...Id]]").`

is this a valid use case? what am I doing wrong?

thanks

Version 13.5.2 (tested with 14.1.0 as doesnt work neither)

**Comment by nirvikpurkait** [comment]
Maybe that's because, you are using optional catch-all route in `[userId]/@sidebar/notifications/[[...id]]/page.tsx` it can resolve to `123/notification/456` and `123/notification` both of them. And again you have `[userId]/notifications/[id]/page.tsx` it can also resolve to `123/notification/456`. As `[userId]/@sidebar/notifications/[[...id]]/page.tsx` and `[userId]/notifications/[id]/page.tsx`, both resolves to the same path it gets confused between which route to show.

Explain: If we consider each route resolving to be secificity of 1,
  - for `[userId]/@sidebar/notifications/[[...id]]/page.tsx` route we have 1 for `[userId]`, 1 for `notifications` and 1 for `[[...id]]`, total 3.
  - and for `[userId]/notifications/[id]/page.tsx`  route we also have 1 for `[userId]`, 1 for `notifications` and 1 for `[id]`, total 3.

As these two different route resolve to the same specificity/path, thus gives the error.

**Reply by sergio-milu** [reply]
I understand your point, thing is that @sidebar/notifications and /notifications are rendered at the same time because I'm using them as parallel routers.

my layout.tsx looks like this

```
<>
    {sidebar}
    {children}
</>
```

so I thought that is a legit use case cause both of them should be shown at the same time
